smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -153,6 +156,33 @@ private void generateCommandMiddlewareResolver(String configType) {
             writer.write("\nconst stack = clientStack.concat(this.middlewareStack);\n");
             writer.openBlock("const handlerExecutionContext: HandlerExecutionContext = {", "}", () -> {
                 writer.write("logger: {} as any,");
+
+                String serviceId = service.getTrait(ServiceTrait.class).map(ServiceTrait::getSdkId).orElse("");
+                writer.write("clientName: '$L',", Arrays.asList(serviceId.split(" "))
+                        .stream().map(StringUtils::capitalize).collect(Collectors.joining("")));
+
+                String operationName = StringUtils.uncapitalize(
+                    symbolProvider.toSymbol(operation).getName().replaceAll("Command$", "")
+                );
+                writer.write("clientName: '$L',", operationName);
+
+                writer.openBlock("inputFilterLog: ", ",", () -> {
+                    Optional<StructureShape> inputShape = operationIndex.getInput(operation);
```

Comment:
This can be simplified by using `OptionalUtils.ifPresentOrElse` (similar change in output section):

```
OptionalUtils.ifPresentOrElse(operationIndex.getInput(operation),
        input -> writer.writeInline("$T.filterSensitiveLog", symbolProvider.toSymbol(input)),
        () -> writer.writeInline("(input) => input"));
```